ros/src/twist_controller/dbw_node.py

```diff
@@ -27,72 +27,129 @@
+        self.pose = None
+        self.velocity = None
+        self.twist = None
+
+        # read ros parameters
+        # vehicle_mass = rospy.get_param('~vehicle_mass', 1736.35)
+        # fuel_capacity = rospy.get_param('~fuel_capacity', 13.5)
+        # brake_deadband = rospy.get_param('~brake_deadband', .1)
+        # wheel_radius = rospy.get_param('~wheel_radius', 0.2413)
+        decel_limit = rospy.get_param('~decel_limit', -5)
+        accel_limit = rospy.get_param('~accel_limit', 1.)
+        wheel_base = rospy.get_param('~wheel_base', 2.8498)
+        steer_ratio = rospy.get_param('~steer_ratio', 14.8)
+        max_lat_accel = rospy.get_param('~max_lat_accel', 3.)
+        max_steer_angle = rospy.get_param('~max_steer_angle', 8.)
 
         self.steer_pub = rospy.Publisher('/vehicle/steering_cmd',
-                                         SteeringCmd, queue_size=1)
+                                         SteeringCmd,
+                                         queue_size=1)
         self.throttle_pub = rospy.Publisher('/vehicle/throttle_cmd',
-                                            ThrottleCmd, queue_size=1)
+                                            ThrottleCmd,
+                                            queue_size=1)
         self.brake_pub = rospy.Publisher('/vehicle/brake_cmd',
-                                         BrakeCmd, queue_size=1)
-
-        # TODO: Create `TwistController` object
-        # self.controller = TwistController(<Arguments you wish to provide>)
-
-        # TODO: Subscribe to all the topics you need to
+                                         BrakeCmd,
+                                         queue_size=1)
+
+        # Create `TwistController` object
+        controller_args = {'min_acc': decel_limit,
+                           'max_acc': accel_limit,
+                           'wheel_base': wheel_base,
+                           'steer_ratio': steer_ratio,
+                           'max_lat_acc': max_lat_accel,
+                           'max_steer_angle': max_steer_angle
+                          }
+        self.controller = Controller(**controller_args)
+
+        # Subscribe to all the topics you need to
+        rospy.Subscriber('/vehicle/dbw_enabled', Bool,
+                         self.dbw_cb, queue_size=1)
+        rospy.Subscriber('/final_waypoints', Lane,
+                         self.waypoints_cb, queue_size=1)
+        rospy.Subscriber('/current_pose', PoseStamped,
+                         self.pose_cb, queue_size=1)
+        rospy.Subscriber('/current_velocity', TwistStamped,
+                         self.velocity_cb, queue_size=1)
+        rospy.Subscriber('/twist_cmd', TwistStamped,
+                         self.twist_cb, queue_size=1)
 
         self.loop()
 
     def loop(self):
-        #pylint: disable=no-self-use
         """
-        STUB
+        The main functionallity of the DBW Node.
+        It runs at 50Hz = 50 cycles per second, until ROS shuts down.
```

Comment:
runs at 10Hz after most recent change

Reply:
I guess this is not fixed. I just changed that for now, for my VM. 
Will probably change later

Reply:
If you run it on a native installation, you can change it back to 50 (i guess) and see if it works